src/jit/lsra.cpp

```diff
@@ -5691,7 +5704,9 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
         }
         else
         {
+#ifndef _TARGET_ARM_
             assert((bestScore & VALUE_AVAILABLE) == 0);
+#endif
```

Comment:
I don't understand why this is not applicable to `_TARGET_ARM_` now. If a constant is being reused, its assigned interval should have caused the previous condition to fire.